src/cephadm/tests/test_cephadm.py

```diff
@@ -318,13 +318,17 @@ def test_skip_firewalld(self, _logger, cephadm_fs):
         with pytest.raises(Exception):
             _cephadm.prepare_dashboard(ctx, 0, 0, lambda _, extra_mounts=None, ___=None : '5', lambda : None)
 
-    @mock.patch('cephadm.logger')
-    @mock.patch('cephadm.fetch_custom_config_files')
-    @mock.patch('cephadm.get_container')
-    def test_to_deployment_container(self, _get_container, _get_config, _logger):
+    def test_to_deployment_container(self, funkypatch):
         """
         test to_deployment_container properly makes use of extra container args and custom conf files
         """
+        from cephadmlib.deployment_utils import to_deployment_container
```

Comment:
Cleanliness and logical locality only.

Reply:
I remembered another reason I like to sometimes put `import`s directly in a test function, *specifically in test code* because if something goes wrong with the import (say a broken dependency) then those tests and only those tests that have the import line fail rather than having pytest unable to execute any test in that module.